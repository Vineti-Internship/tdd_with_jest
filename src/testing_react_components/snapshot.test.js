import React from 'react';
import { mount, shallow } from 'enzyme';
import Parent from './stateful';

describe('A component', () => {
  let component;
  describe('parent component', () => {
    // With enzyme
    test('renders one child component', () => {});
    test('renders one button', () => {});

    test('matches snapshot', () => {
      component = mount(<Parent />);
      expect(component).toMatchSnapshot();
    });

    test('matches snapshot', () => {
      component = shallow(<Parent />);
      expect(component).toMatchSnapshot();
    });
  });
});