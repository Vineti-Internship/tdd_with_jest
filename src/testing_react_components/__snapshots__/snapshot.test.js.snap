// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A component parent component matches snapshot 1`] = `
<Parent>
  <Child
    n={0}
  >
    <h1>
      0
    </h1>
  </Child>
  <button
    onClick={[Function]}
  >
    Click
  </button>
</Parent>
`;

exports[`A component parent component matches snapshot 2`] = `
<Fragment>
  <Child
    n={0}
  />
  <button
    onClick={[Function]}
  >
    Click
  </button>
</Fragment>
`;
